gio/src/unix_fd_list.rs

```diff
@@ -54,37 +54,41 @@ impl<O: IsA<UnixFDList>> UnixFDListExtManual for O {
         }
     }
 
-    fn get(&self, index_: i32) -> Result<RawFd, glib::Error> {
+    fn get(&self, index_: i32) -> Result<OwnedFd, glib::Error> {
         unsafe {
             let mut error = ptr::null_mut();
-            let ret = ffi::g_unix_fd_list_get(self.as_ref().to_glib_none().0, index_, &mut error);
+            let raw_fd =
+                ffi::g_unix_fd_list_get(self.as_ref().to_glib_none().0, index_, &mut error);
+            let fd = OwnedFd::from_raw_fd(raw_fd);
             if error.is_null() {
-                Ok(ret)
+                Ok(fd)
             } else {
                 Err(from_glib_full(error))
             }
         }
     }
 
-    fn peek_fds(&self) -> Vec<RawFd> {
+    fn peek_fds(&self) -> Vec<BorrowedFd<'_>> {
         unsafe {
             let mut length = mem::MaybeUninit::uninit();
-            let ret = FromGlibContainer::from_glib_none_num(
+            let ret: Vec<RawFd> = FromGlibContainer::from_glib_none_num(
                 ffi::g_unix_fd_list_peek_fds(self.as_ref().to_glib_none().0, length.as_mut_ptr()),
                 length.assume_init() as usize,
             );
-            ret
+            ret.into_iter()
+                .map(|fd| BorrowedFd::borrow_raw(fd))
+                .collect()
         }
     }
 
-    fn steal_fds(&self) -> Vec<RawFd> {
+    fn steal_fds(&self) -> Vec<OwnedFd> {
         unsafe {
             let mut length = mem::MaybeUninit::uninit();
-            let ret = FromGlibContainer::from_glib_full_num(
+            let ret: Vec<RawFd> = FromGlibContainer::from_glib_full_num(
                 ffi::g_unix_fd_list_steal_fds(self.as_ref().to_glib_none().0, length.as_mut_ptr()),
                 length.assume_init() as usize,
             );
-            ret
+            ret.into_iter().map(|fd| OwnedFd::from_raw_fd(fd)).collect()
```

Comment:
Why so complicated? Implement `glib::translate::TransparentType for OwnedFd` (in glib/src/translate.rs), and then you can do `glib::Slice::from_glib_full_num()` from the return value directly without all the vecs

Reply:
https://gtk-rs.org/gtk-rs-core/git/docs/glib/translate/trait.TransparentType.html

Reply:
It requires Clone, which OwnedFd does not have.

Reply:
Ah I see, that's sad. I wonder why it doesn't have clone, that's just `dup()`.

Reply:
well, according to rust, duplicating  a fd can fail, https://doc.rust-lang.org/src/std/os/fd/owned.rs.html#94.